t/20_sdaf_deployment_library.t

```diff
@@ -547,4 +565,85 @@ subtest '[convert_region_to_short] Test invalid input' => sub {
     dies_ok { convert_region_to_short($_) } "Croak with invalid region name: $_" foreach @invalid_region_names;
 };
 
+subtest '[sdaf_execute_playbook] Fail with missing mandatory arguments' => sub {
+    my $ms_sdaf = Test::MockModule->new('sles4sap::sdaf_deployment_library', no_auto => 1);
+    my $croak_message;
+    $ms_sdaf->redefine(croak => sub { $croak_message = $_[0]; die(); });
+
+    set_var('SDAF_ENV_CODE', 'LAB');
+    set_var('SDAF_WORKLOAD_VNET_CODE', 'SAP04');
+    set_var('PUBLIC_CLOUD_REGION', 'swedencentral');
+    set_var('SAP_SID', 'QES');
+
+    dies_ok { sdaf_execute_playbook() } 'Croak with missing mandatory argument "playbook_filename"';
+    ok $croak_message =~ m/playbook_filename/, 'Check if failure is triggered at correct place.';
+
+    undef_variables();
+};
+
+subtest '[sdaf_execute_playbook] Command execution' => sub {
+    my $ms_sdaf = Test::MockModule->new('sles4sap::sdaf_deployment_library', no_auto => 1);
+    my @calls;
+
+    $ms_sdaf->redefine(upload_logs => sub { return; });
+    $ms_sdaf->redefine(log_dir => sub { return '/tmp/openqa_logs'; });
+    $ms_sdaf->redefine(assert_script_run => sub { return 0; });
+    $ms_sdaf->redefine(deployment_dir => sub { return '/tmp/SDAF'; });
+    $ms_sdaf->redefine(record_info => sub { return; });
+    $ms_sdaf->redefine(script_run => sub { return; });
```

Comment:
Functional it is perfectly the same but I suggest to collect the call in script_run and not in log_command_output, 

```
$ms_sdaf->redefine(script_run => sub {  push(@calls, $_[1]); return;  });
$ms_sdaf->redefine(log_command_output => sub { return $_[1]; });
```

Reply:
I was trying to avoid script_run since I would have to filter out the 'chmod' commands, but since those are going to be moved elsewhere I can do that easily. 
Will be back with fix.

Reply:
Fixed. Thanks for suggestion.

Reply:
Fixed. Thanks for suggestion.